### PR TITLE
Adding BRS for app service end to end tls

### PR DESCRIPTION
**Pull request by cbulbule**

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1 [hunk 1]
```diff
@@ -0,0 +1,943 @@
...
+
+        3. To enable End-to-end TLS encryption on all App Services in a Subscription, from a previously taken snapshot:
+           Enable-EndToEndTLSForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\Enable-EndToEndTLSForAppServices\AppServicesWithNonCompliant.csv
+
+        To know more about parameter execute:
+            Get-Help Enable-EndToEndTLSForAppServices -Detailed
+
+    To roll back:
+        1. To configure AllAllowed on the production slot and all non-production slots of all App Services in a Subscription, from a previously taken snapshot:
+           Disable-EndToEndTLSForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\DisableEndToEndTLSForAppServices\RemediatedForAppServices.csv
```
Comment (Aboli-msft) [inline Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1:51]: RemediatedForAppServices--> RemediatedAppService

File: Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1 [hunk 2]
```diff
@@ -0,0 +1,943 @@
...
+       
+        To know more about the options supported by the roll back command, execute:
+        Get-Help Disable-EndToEndTLSForAppServices -Detailed
```
Comment (Aboli-msft) [inline Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1:54]: should be Enable-EndToEndTLSForAppServices

File: Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1 [hunk 3]
```diff
@@ -0,0 +1,943 @@
...
+        {
+            Write-Host "No App Services found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        $validResources | ForEach-Object { 
+            try
+            {
+                $appServiceResourceAPI = "https://management.azure.com/subscriptions/$subId/resourceGroups/$($_.ResourceGroupName)/providers/Microsoft.Web/sites/$($_.Name)?api-version=2024-04-01" 
+                $AppServiceDetail = Fetch-API -Method Get -Uri $appServiceResourceAPI -ErrorAction Stop
```
Comment (Aboli-msft) [inline Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1:275]: SHould not we filter for applicable kinds?

File: Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1 (hunk 2 above)
Comment (cbulbule) [reply Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1:54]: we are executing Disable-EndToEndTLSForAppServices for roll back not for enable it

File: Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1 [hunk 4]
```diff
@@ -0,0 +1,943 @@
...
+# Examples:
+    To remediate:
+        1. To review the App Services in a Subscription that will be remediated:
+           Enable-EndToEndTLSForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
+
+        2. To enable End-to-end TLS encryption on all App Services in a Subscription:
+           Enable-EndToEndTLSForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck
+
+        3. To enable End-to-end TLS encryption on all App Services in a Subscription, from a previously taken snapshot:
+           Enable-EndToEndTLSForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\Enable-EndToEndTLSForAppServices\AppServicesWithNonCompliant.csv
```
[the remark this answers is not in the document]
Comment (cbulbule) [reply Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1]: Done

File: Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1 (hunk 3 above)
Comment (cbulbule) [reply Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1]: Done

File: Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1 (hunk 1 above)
Comment (cbulbule) [reply Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1]: Done

File: Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1 [hunk 5]
```diff
@@ -0,0 +1,927 @@
...
+    }
+    else
+    {
+        # No file path provided as input to the script. Fetch all App Services in the Subscription.
+        if ([String]::IsNullOrWhiteSpace($FilePath))
+        {
+            try
+            {
+                Write-Host "Fetching all App Services in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+
```
Comment (rajatgoyalMS) [inline Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1:306]: image
Can you take reference from the above image of BRS and make changes accordingly?

Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1:306]: For reference - 
https://github.com/azsk/AzTS-docs/pull/455

Comment (cbulbule) [reply Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1:306]: Done

File: Scripts/RemediationScripts/ControlsEligibleForRemediationThroughUI.md [hunk 6]
```diff
@@ -81,6 +81,8 @@
 
 40. [Azure_ServiceBus_AuthN_Disable_Local_Auth](ControlsEligibleForRemediationThroughUI.md#40-azure_servicebus_authn_disable_local_auth)
 
+41. [Azure_AppService_DP_Configure_EndToEnd_TLS](ControlsEligibleForRemediationThroughUI.md#40-Azure_AppService_DP_Configure_EndToEnd_TLS)
```
Comment (rajatgoyalMS) [inline Scripts/RemediationScripts/ControlsEligibleForRemediationThroughUI.md:84]: This link on the UI on repo is not working?

**Review by Aboli-msft — APPROVED** [review]

File: Scripts/RemediationScripts/ControlsEligibleForRemediationThroughUI.md (hunk 6 above)
Comment (cbulbule) [reply Scripts/RemediationScripts/ControlsEligibleForRemediationThroughUI.md]: Done

**Review by rajatgoyalMS — APPROVED** [review]

**Pull request merged**